cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1050,33 +1063,62 @@ public MapAndStatus(T loadMap, K status) {
                                         + ": No deployed Cruise Control for doing a rebalance.")).mapEmpty();
                     }
 
-                    CruiseControlApi apiClient = cruiseControlClientProvider();
+                    String ccSecretName =  CruiseControlResources.secretName(clusterName);
+                    String ccApiSecretName =  CruiseControlResources.apiSecretName(clusterName);
+                    String coSecretName =  ClusterOperator.secretName(clusterName);
 
-                    // get latest KafkaRebalance state as it may have changed
-                    return kafkaRebalanceOperator.getAsync(kafkaRebalance.getMetadata().getNamespace(), kafkaRebalance.getMetadata().getName())
-                        .compose(currentKafkaRebalance -> {
-                            KafkaRebalanceStatus kafkaRebalanceStatus = currentKafkaRebalance.getStatus();
-                            KafkaRebalanceState currentState;
-                            // cluster rebalance is new or it is in one of the others states
-                            if (kafkaRebalanceStatus == null || kafkaRebalanceStatus.getConditions().stream().filter(cond -> "ReconciliationPaused".equals(cond.getType())).findAny().isPresent()) {
-                                currentState = KafkaRebalanceState.New;
-                            } else {
-                                String rebalanceStateType = rebalanceStateConditionType(kafkaRebalanceStatus);
-                                if (rebalanceStateType == null) {
-                                    throw new RuntimeException("Unable to find KafkaRebalance State in current KafkaRebalance status");
+                    Future<Secret> ccSecretFuture = secretOperations.getAsync(clusterNamespace, ccSecretName);
+                    Future<Secret> ccApiSecretFuture = secretOperations.getAsync(clusterNamespace, ccApiSecretName);
+                    Future<Secret> coSecretFuture = secretOperations.getAsync(clusterNamespace, coSecretName);
```

Comment:
How often does this loop run? I remember in the past that the rebalance assembly was quite aggressive in reconciling. You should make sure to get the secrets only once and not once per second (assuming this is the loop which runs in very short intervals).

Reply:
Once every two minutes. It also gets hit once after a user updates the KafkaRebalance resource. Is this infrequent enough to leave here? Or do you still think I should refactor?

Reply:
If it is really only every 2 minutes it should be ok. But I know that at some point the KafkaRebalanceAssemblyOperator was running a separate reconciliation loop which was running every second or so. That would be too often for sure.